lib/controllers/users.js

```diff
@@ -0,0 +1,50 @@
+const { Router } = require('express');
+const authenticate = require('../middleware/authenticate');
+const authorize = require('../middleware/authorize');
+const User = require('../models/User');
+const UserService = require('../services/UserService');
+const ONE_DAY_IN_MS = 1000 * 60 * 60 * 24;
+
+module.exports = Router()
+  .post('/', async (req, res, next) => {
+    try {
+      const user = await UserService.create(req.body);
+      res.json(user);
+    } catch (e) {
+      next(e);
+    }
+  })
+
+  .post('/sessions', async (req, res, next) => {
+    try {
+      const { email, password } = req.body;
+      const sesToken = await UserService.signIn({ email, password });
+      res
+        .cookie(process.env.COOKIE_NAME, sesToken, {
+          httpOnly: true,
+          maxAge: ONE_DAY_IN_MS,
+        })
+        .json({ message: 'Welcome you are signed in!' });
+    } catch (e) {
+      next(e);
+    }
+  })
+
+  .get('/me', authenticate, async (req, res) => {
+    res.json(req.user);
+  })
+
+  .get('/', [authenticate, authorize], async (req, res, next) => {
+    try {
+      const users = await User.getAll();
+      res.send(users);
+    } catch (e) {
+      next(e);
+    }
+  })
```

Comment:
These two routes are not needed for this lab, but it is good practice to add in authenticate and authorize middleware.